tasks/ui_framework.js

```diff
@@ -6,6 +6,37 @@ import debounce from 'lodash/function/debounce';
 const platform = require('os').platform();
 
 module.exports = function (grunt) {
+  grunt.registerTask('uiFramework:build', function () {
+    const done = this.async();
+
+    const serverCmd = {
+      cmd: /^win/.test(platform) ? '.\\node_modules\\.bin\\webpack.cmd' : './node_modules/.bin/webpack',
+      args: [
+        '-p',
+        '--config=ui_framework/doc_site/webpack.config.js',
+      ],
+      opts: { stdio: 'inherit' }
+    };
+
+    const uiFrameworkServerBuild = new Promise((resolve, reject) => {
```

Comment:
Yup:

```javascript
grunt.registerTask('uiFramework:build', function () {
    const serverCmd = {
      cmd: isPlatformWindows ? '.\\node_modules\\.bin\\webpack.cmd' : './node_modules/.bin/webpack',
      args: [
        '-p',
        '--config=ui_framework/doc_site/webpack.config.js',
      ],
      opts: { stdio: 'inherit' }
    };

    return new Promise((resolve, reject) => {
      grunt.util.spawn(serverCmd, (error, result, code) => {
        if (error || code !== 0) {
          const message = result.stderr || result.stdout;

          grunt.log.error(message);

          return reject();
        }

        grunt.log.writeln(result);

        resolve();
      });
    });
  });
```

I believe you need to use `this.aync()` to tell grunt that the process can be killed.